packages/flutter/test/widgets/box_decoration_test.dart

```diff
@@ -27,20 +32,25 @@ class TestImageProvider extends ImageProvider<TestImageProvider> {
 
   @override
   ImageStreamCompleter loadImage(TestImageProvider key, ImageDecoderCallback decode) {
+    assert(_images.isNotEmpty, 'ask for more images in `prepareImages`');
+    final ui.Image image = _images.last;
+    _images.removeLast();
+
     return OneFrameImageStreamCompleter(
-      future.then<ImageInfo>((void value) => ImageInfo(image: image)),
+      future.then<ImageInfo>((void value) {
+        final ImageInfo result = ImageInfo(image: image);
+        return result;
+      }),
     );
   }
 }
 
 Future<void> main() async {
   AutomatedTestWidgetsFlutterBinding();
-  TestImageProvider.image = await decodeImageFromList(Uint8List.fromList(kTransparentImage));
+  await TestImageProvider.prepareImages(2);
 
-  testWidgets('DecoratedBox handles loading images',
-  // TODO(polina-c): dispose ImageStreamCompleterHandle, https://github.com/flutter/flutter/issues/145599 [leaks-to-clean]
-  experimentalLeakTesting: LeakTesting.settings.withIgnoredAll(),
-  (WidgetTester tester) async {
+  testWidgets('DecoratedBox handles loading images', (WidgetTester tester) async {
+    addTearDown(imageCache.clear);
```

Comment:
It is a bit of a smell that you have to manually clear the cache in tests. The cache should be more or less transparent.

Reply:
I would say it is more for test isolation, than for functionality. The cache is transparent for functionality.